tm.plugin.factiva/R/readFactivaXML.R

```diff
@@ -14,11 +14,9 @@ readFactivaXML <- readXML(spec = list(
     id = list("function", function(node) {
               str <- gsub("^distdoc:archive/ArchiveDoc::Article/", "",
                           xml_text(xml_find_all(node, ".//reference")))
-              # Extract useful information: origin, date, and two last characters to avoid collisions
-              m <- regmatches(str, regexec("^([A-Za-z]+)0*[1-9][0-9]([0-9][0-9][0-3][0-9][0-3][0-9]).*([A-Za-z0-9]{2})$", str))[[1]]
               # If extraction failed for some reason, make sure we return a unique identifier
-              if(length(m) == 4)
-                  paste(toupper(m[2]), "-", m[3], "-", m[4], sep="")
+              if(nchar(str) > 0)
```

Comment:
No, but it's a good question. The predecessor code is:
```
              str <- gsub("^distdoc:archive/ArchiveDoc::Article/", "",
                          xml_text(xml_find_all(node, ".//reference")))
```
If `xml_find_all` finds nothing, it will return an `{xml_nodeset (0)}`, meaning `xml_text` and subsequently `gsub` will return `character(0)`, then `nchar(str)` will return `integer(0)`, then the `if` statement will fail with "argument is of length zero".

I think `if(length(str) > 0 && nchar(str) > 0` would do it.